packages/flutter/lib/src/widgets/two_dimensional_viewport.dart

```diff
@@ -1304,23 +1356,59 @@ abstract class RenderTwoDimensionalViewport extends RenderBox implements RenderA
 
   void _insertChild(RenderBox child, ChildVicinity slot) {
     assert(_debugTrackOrphans(newOrphan: _children[slot]));
+    assert(!_keepAliveBucket.containsValue(child));
```

Comment:
also check _debugDanglingKeepAlives ?

Reply:
Should it be empty, or not contain the child? We do not have a similar check in RenderSliverMultiBoxAdaptor, should I add one there too? RenderTwoDimensionalViewport is like a cross between RenderSliverMultiBoxAdaptor and Viewport with two offsets, so I have been trying to match behavior as much as possible for consistency. :)

Reply:
I think they should not contain the child because it should be removed from the _debugDanglingKeepAlives.

Given this is mirroring the 1D case, I am fine either you add both or not.

Can also think about extracting a mixin with generic slot type if things start to get repeated.